zenpacklib.py

```diff
@@ -621,9 +624,23 @@ def _create_indexes(cls, zcatalog, spec):
                 # reindex all objects of this type so they are added to the
                 # catalog.
                 results = ICatalogTool(context).search(types=(classname,))
+
+                # global catalog (I think) for removing bad entries
+                zcat = context.getDmd().global_catalog
+
                 for result in results:
-                    if hasattr(result.getObject(), 'index_object'):
-                        result.getObject().index_object()
+                    try:
+                        ob = result.getObject()
+                        if hasattr(ob, 'index_object'):
+                            ob.index_object()
+                    except Exception:
+                        LOG.error('Problem indexing bad catalog entry: %s' % result.getPath())
+                        # not sure if this is appropriate to do here or not.
+                        # but these bad paths didn't show up in dmd.global_catalog
+                        # without using ICatalogTool
+                        if not leaveObjects:
+                            LOG.info('Removing bad catalog entry: %s' % result.getPath())
+                            zcat.uncatalog_object(result.getPath())
```

Comment:
I vote for not doing this at all, even with the option.  We don't generally do it elsewhere in our code- I think the assumption is that if your catalog is showing stale entries, we want you to get out the toolbox and scan it properly.